topics/digital-humanities/tutorials/open-refine-tutorial/tutorial.md

```diff
@@ -0,0 +1,473 @@
+
+## Remove blank rows
+
+> <hands-on-title>Removing the blank rows</hands-on-title>
+>
+> 1. Click on the triangle on the left of `Record ID`.  
+>
+>    <p align="center">
+>      <img src="sort.png" alt="Sort Record ID">
+>    </p>
+>
+> 2. Click on `Sort...`.  
+>
+> 3. Select `numbers` and click on `OK`.  
+>
+>    <p align="center">
+>      <img src="sort2.png" alt="Sort Record ID options">
+>    </p>
+>
+> 4. Above the table, click on `Sort` and select `Reorder rows permanently`.  
+>
+>    <p align="center">
+>      <img src="sort3.png" alt="Sort Record ID reorder permanently">
+>    </p>
+>
+> 5. Click on the triangle left of the `Record ID` column. Hover over `Edit cells` and select `Blank down`.  
+>
+>    <p align="center">
+>      <img src="sort4.png" alt="Blank down Record ID">
+>    </p>
+>
+> 6. Click on the triangle left of the `Record ID` column. Hover over `Facet`, then move your mouse to `Customized facets` and select `Facet by blank (null or empty string)`.  
+>
+>    <p align="center">
+>      <img src="sort5.png" alt="Facet by blank Record ID">
+>    </p>
+>
+> 7. On the left, a new option appears under `Facet/Filter` with the title `Record ID`. Click on `true`.  
+>
+>    <p align="center">
+>      <img src="sort6.png" alt="Facet by blank true Record ID">
+>    </p>
+>
+> 8. Click on the triangle to the left of the column called `All`. Hover over `Edit rows`, and select `remove matching rows`.  
+>
+>    <p align="center">
+>      <img src="deduplicate.png" alt="Remove matching rows Record ID">
+>    </p>
+>
+> 9. Close the `Facet` by clicking on the cross (x) to see all rows.  
+>
+{: .hands_on}
+
+> <question-title></question-title>
+>
+> 1. How many rows have been removed?
+>
+> > <solution-title></solution-title>
+> >
+> > 1. 84
+> >
+> {: .solution}
+{: .question}
+
+The dataset does not contain any more blank rows now. But we need to do more cleaning to improve the dataset.
+
+## Use GREL
+
+> <hands-on-title>Find and replace typos using GREL</hands-on-title>
+>
+> To remove the occurance of double pipe \|\| from the file we can do the following:
+> 1. Click on the triangle on the left of `Categories` and select `Text filter`.
+> 2. On the left, using the `Facet/Filter` section, search for the occurrence of \| and \|\|. There are 71061 rows with \| and 9 rows with \|\|. We want to remove these 9 lines as they are there by mistake.
+> 3. Click on the triangle on the left of `Categories`, hover over `edit cells`, and click on `Transform...`.
+> 4. In the new window, use the following text `value.replace('||', '|')` as "Expression" and click on `OK`.
+>
+>    <p align="center">
+>      <img src="filter_grel3.png" alt="Custom text transform on column Categories">
+>    </p>
+>
+> We can also remove the double occurrence of the same for different entries as follows:
+> 1. Click on the triangle on the left of `Categories`, hover over `edit cells`, and click on `Transform...`.
```

Comment:
to keep the numbering going througout the entire hands-on box, you can change the indentation like this:

```suggestion
>    We can also remove the double occurrence of the same for different entries as follows:
> 5. Click on the triangle on the left of `Categories`, hover over `edit cells`, and click on `Transform...`.
```